Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc

```diff
@@ -183,36 +171,74 @@ export function runQTest(args: QTestArguments): Result {
-                additionalTempDirectories : [sandboxDir],
-                privilegeLevel: args.privilegeLevel,
-                dependencies: [
-                    //When there are test failures, and PDBs are looked up to generate the stack traces,
-                    //the original location of PDBs is used instead of PDBs in test sandbox. This is 
-                    //a temporary solution until a permanent fix regarding the lookup is identified
-                    ...(args.qTestInputs ? args.qTestInputs.filter(
-                        f => f.name.hasExtension && f.name.extension === a`.pdb`
-                    ) : []),
-                    ...(args.qTestRuntimeDependencies || []),
-                ],
-            }, 
-            untrackingCBPaths
-        )
-    );
+    let unsafeOptions = {
+        untrackedPaths: [
+            qTestContextInfoPath,
+        ],
+        untrackedScopes: [
+            d`d:/data`,
+            d`d:/app`,
+            ...addIf(Environment.hasVariable("QAUTHMATERIALROOT"), Environment.getDirectoryValue("QAUTHMATERIALROOT")),
+        ]
+    };
+
+    let result = Transformer.execute({
+        tool: args.qTestTool ? args.qTestTool : qTestTool,
+        tags: tags,
+        description: args.description,
+        arguments: commandLineArgs,
+        consoleOutput: consolePath,
+        workingDirectory: sandboxDir,
+        tempDirectory: tempDirectory,
+        weight: args.weight,
+        environmentVariables: [{ name: "[Sdk.BuildXL]qCodeCoverageEnumType", value: qCodeCoverageEnumType }],
+        disableCacheLookup: Environment.getFlag("[Sdk.BuildXL]qTestForceTest"),
+        additionalTempDirectories : [sandboxDir],
+        privilegeLevel: args.privilegeLevel,
+        dependencies: [
+            //When there are test failures, and PDBs are looked up to generate the stack traces,
+            //the original location of PDBs is used instead of PDBs in test sandbox. This is
+            //a temporary solution until a permanent fix regarding the lookup is identified
+            ...(args.qTestInputs ? args.qTestInputs.filter(
+                f => f.name.hasExtension && f.name.extension === a`.pdb`
+            ) : []),
+        ],
+        unsafe: unsafeOptions
+    });
+
     const qTestLogsDir: StaticDirectory = result.getOutputDirectory(logDir);
+
+    // If code coverage is enabled, schedule a pip that will perform coverage file upload.
+    if (qCodeCoverageEnumType === "DynamicCodeCov"){
+        const parentDir = d`${logDir}`.parent;
+        const leafDir = d`${logDir}`.nameWithoutExtension;
+        const coverageLogDir = d`${parentDir}/CoverageLogs/${leafDir}`;
+        const coverageConsolePath = p`${coverageLogDir}/coverageUpload.stdout`;
+
+        const commandLineArgsForUploadPip: Argument[] = [
+            Cmd.option("--qTestLogsDir ", Artifact.output(coverageLogDir)),
+            Cmd.option("--qTestContextInfo ", Artifact.input(qTestContextInfoPath)),

```

Comment:
Can you remove Artifact.input for qtestcontextinfopath? this will introduce an error complaining the path is in an untracked mount.